spaces/views.py

```diff
@@ -150,6 +150,37 @@ def space_members(request, pk):
 
 @login_required
 def my_spaces_list(request):
+    interest_list = [
+        "Automotive",
+        "Beauty",
+        "Book and Literature",
+        "Business",
+        "Education",
+        "Events",
+        "Gaming",
+        "Health",
+        "Food and drink",
+        "Politics",
+        "Movies",
+        "Music",
+    ]
+
+    # Create or get existing interests
+    for interest in interest_list:
+        Interest.objects.get_or_create(name=interest)
+
+    all_interests = Interest.objects.all()
+
+    if request.method == "POST":
+        selected_interest = request.POST.get("interests")
+        try:
+            selected_interest_obj = Interest.objects.get(pk=selected_interest)
+            interest_spaces = Space.objects.filter(interests=selected_interest_obj)
+        except Interest.DoesNotExist:
+            interest_spaces = Space.objects.none()
+    else:
+        interest_spaces = Space.objects.none()
+
     spaces = Space.objects.filter(
```

Comment:
Previous line states Space.objects.none but this line assigns spaces. So what is the logic here? Can you maybe also add request.method == GET to separate the logic here?

Reply:
The Space.objects.none() statement is used to create an empty QuerySet. This is used as a placeholder for interest_spaces in case the Interest object does not exist or if the request method is not "POST". Separating the logic for "GET" and "POST" can certainly make the code clearer I will be working on it. Thanks for comments.